api/circle/circle.spec.js

```diff
@@ -0,0 +1,55 @@
+require('chai').should();
+
+const app = require('../../app');
+
+const expect = require('chai').expect;
+require('chai').should();
+
+const request = require('supertest');
+
+
+const circleDAO = require('../../dao/circle');
+
+describe('/circle api', () => {
+  let circleId;
+  it('it should create a new circle', (done) => {
```

Comment:
Add pre-run assertions for all test cases.

i.e., we also have to assert that the circle wasn't existing before calling the create-circle API.